Add recipe for python-tdd-mode package

marcwebbie opened the pull request:
### Brief summary of what the package does

[python-tdd-mode](https://github.com/marcwebbie/python-tdd-mode) is a lightweight, modern Emacs minor mode designed for Python developers practicing test-driven development (TDD)

### Direct link to the package repository

https://github.com/marcwebbie/python-tdd-mode

### Your association with the package

Maintainer

### Relevant communications with the upstream package maintainer

None needed

### Checklist

<!-- Please confirm by replacing `[]` with `[x]`: -->

- [X] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses)
- [X] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [X] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [X] My elisp byte-compiles cleanly
- [X] I've used `M-x checkdoc` to check the package's documentation strings
- [X] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)

<!-- After submitting, please fix any problems the CI reports. -->

riscy commented:
Thanks for this. Quick first pass -

⸺ `tdd-mode.el` with byte-compile using Emacs 30.1:
```
tdd-mode.el:494:2: Warning: docstring wider than 80 characters
```

⸺ `tdd-mode.el` with [melpazoid](https://github.com/riscy/melpazoid):
```
- tdd-mode.el#L180: It's safer to sharp-quote function names; use `#'`
- tdd-mode.el#L222: It's safer to sharp-quote function names; use `#'`
- tdd-mode.el#L432: It's safer to sharp-quote function names; use `#'`
- tdd-mode.el#L377: It's safer to sharp-quote function names; use `#'`
- tdd-mode.el#L525: It's safer to sharp-quote function names; use `#'`
- tdd-mode.el#L529: It's safer to sharp-quote function names; use `#'`
- tdd-mode.el#L178: Only use back/front quotes to link to top-level elisp symbols
- tdd-mode.el#L178: Only use back/front quotes to link to top-level elisp symbols
- tdd-mode.el#L183: Only use back/front quotes to link to top-level elisp symbols
- tdd-mode.el#L191: Only use back/front quotes to link to top-level elisp symbols
- tdd-mode.el#L219: Only use back/front quotes to link to top-level elisp symbols
- tdd-mode.el#L347: Only use back/front quotes to link to top-level elisp symbols
- tdd-mode.el#L361: Only use back/front quotes to link to top-level elisp symbols
- tdd-mode.el#L332: You may want to consider `(derived-mode-p 'xyz)`
- tdd-mode.el#L163: Top-level `define-key` can overwrite bindings.  Try: `(defvar my-map (let ((km (make-sparse-keymap))) (define-key ...) km))`
```


⸺ Package and license:
- Prefer default recipe: `(tdd-mode :fetcher github :repo "marcwebbie/tdd-mode")`
- Try to use a standard license file format for your repo
  This helps detection tools like: https://github.com/licensee/licensee

marcwebbie commented:
Hi @riscy I updated the package

- [X] Fixed the LICENCE
- [X] Fixed the docstring warning
- [X] Fixed the melpazoid warnings

Thanks for the great work with melpa. :)

marcwebbie commented:
- [X] Updated recipe

riscy commented:
Thank you!  This looks cool, and maybe TDD is going to have a renaissance with people using genAI.

The last thing I wanted to check with you is the name.  I can see this is a Python-focused tdd-mode, and I think it's likely that someone in the future might want to create an elisp-tdd-mode or a ruby-tdd-mode or a rust-tdd-mode (etc.); do you have any plans in the future for that?  Or, alternatively, would `python-tdd-mode` be a more apt name?

riscy commented:
> It will be a language agnostic package

I see "it will be" a lot when reviewing packages, so I have some small advice here: "language agnostic" is probably an order of magnitude more work than targeting a single language. Maybe you realize this already!  But until that time comes, my feeling is that calling this `python-tdd-mode` is be the safer bet.

Later on when you create a TDD mode for all languages, or even multiple popular languages, `tdd-mode` makes sense - and we have tools we can use to make the migration easy for users - or you can just set up a dependency tree.

riscy commented:
(Let me know what you think about the rename.)

marcwebbie commented:
Hi @riscy appologies for the delay but I followed your idea and renamed the package to python-tdd-mode
